layers/+distributions/spacemacs-base/funcs.el

```diff
@@ -380,16 +380,22 @@ projectile cache when it's possible.
 When ASK-USER is non-nil, user will be asked to confirm file
 removal."
   (interactive "f")
-  (when (and filename (file-exists-p filename))
-    (let ((buffer (find-buffer-visiting filename)))
-      (when buffer
-        (kill-buffer buffer)))
-    (when (or (not ask-user)
-              (yes-or-no-p "Are you sure you want to delete this file? "))
-      (delete-file filename)
+  (let ((is-dir (file-directory-p filename)))
+    (when (and (file-exists-p filename)
+               (or (not ask-user)
+                   (yes-or-no-p
+                    (message  "Are you sure you want to delete this %s? "
+                              (if is-dir "directory" "file")))))
+      (spacemacs/kill-file-or-dir-buffers filename)
+      (if (file-directory-p filename)
+          (delete-directory filename t)
```

Comment:
No why? We don't move files to trash, we remove them, so it's consistent with a line after. 

Though we might add an argument for trashing.